Playground/js/main.js

```diff
@@ -138,23 +138,43 @@ compileAndRun = function(parent, fpsLabel) {
             }
 
             engine = engine;
-            engine.runRenderLoop(function () {
-                if (engine.scenes.length === 0) {
-                    return;
-                }
-
-                if (canvas.width !== canvas.clientWidth) {
-                    engine.resize();
-                }
-
-                var scene = engine.scenes[0];
-
-                if (scene.activeCamera || scene.activeCameras.length > 0) {
-                    scene.render();
-                }
-
-                fpsLabel.innerHTML = engine.getFps().toFixed() + " fps";
-            }.bind(this));
+            var sceneToRender;
+            if (scene.then) {

```

Comment:
No need to dupplicate so we ll check tomorrow together :-)

Reply:
Well this is really important to consider that your code is public and must be clean and up to great standards.